### PR TITLE
Feat/token reuse

### PR DESCRIPTION
**Pull request by Ndiritu**
related to https://github.com/microsoftgraph/msgraph-sdk-php/issues/1407

### CODE REVIEW

**Comment by Ndiritu** [comment]
Unit tests should fail until we release authentication changes https://github.com/microsoft/kiota-authentication-phpleague-php/pull/72

**Review by baywet — DISMISSED** [review]

File: src/Authentication/GraphPhpLeagueAccessTokenProvider.php [hunk 1]
```diff
@@ -61,6 +64,27 @@ public function __construct(
             $tokenBaseServiceUrl,
             $nationalCloud
         );
-        parent::__construct($tokenRequestContext, $scopes, $allowedHosts, $oauthProvider);
+        $offlineAccessEnabled = in_array('offline_access', array_map(fn ($scope) => trim(strtolower($scope)), $scopes), true);
+        if (!$offlineAccessEnabled) {
+            $scopes[] = 'offline_access';
```
Comment (baywet) [inline src/Authentication/GraphPhpLeagueAccessTokenProvider.php:69]: I'm not sure what's the rationale behind adding that scope forcibly?

Comment (Ndiritu) [reply src/Authentication/GraphPhpLeagueAccessTokenProvider.php:69]: Thinking it's safe to always request a refresh token? This covers the case that developers may not be aware of the `offline_access` scope being required.

MSAL actually adds a bunch of other [defaults](https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/bc6ab89c9caec96f312ed2e1d7dc0637a166ea74/msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/OAuthAuthorizationGrant.java#L25)

Comment (baywet) [reply src/Authentication/GraphPhpLeagueAccessTokenProvider.php:70]: I'm not sure all flows support returning a refresh token, I know that implicit flow didn't and the service would barf at you back then.

Comment (Ndiritu) [reply src/Authentication/GraphPhpLeagueAccessTokenProvider.php]: Thanks for context. Reverted this.

File: src/Authentication/GraphPhpLeagueAccessTokenProvider.php [hunk 2]
```diff
@@ -38,13 +38,16 @@ class GraphPhpLeagueAccessTokenProvider extends PhpLeagueAccessTokenProvider
...
      * @param string $nationalCloud Defaults to https://graph.microsoft.com. See
      * https://learn.microsoft.com/en-us/graph/deployments
+     * @param AccessTokenCache|null $accessTokenCache Defaults to an in-memory cache if null
      */
     public function __construct(
         TokenRequestContext $tokenRequestContext,
         array $scopes = [],
-        string $nationalCloud = NationalCloud::GLOBAL
+        string $nationalCloud = NationalCloud::GLOBAL,
+        ?AccessTokenCache $accessTokenCache = null
```
Comment (baywet) [inline src/Authentication/GraphPhpLeagueAccessTokenProvider.php:47]: should we set the default value to our implementation instead?

**Review by baywet — APPROVED** [review]

File: src/Authentication/GraphPhpLeagueAccessTokenProvider.php (hunk 2 above)
Comment (Ndiritu) [reply src/Authentication/GraphPhpLeagueAccessTokenProvider.php:47]: This is set at the Kiota core level in the [auth package](https://github.com/microsoft/kiota-authentication-phpleague-php/blob/dev/src/PhpLeagueAccessTokenProvider.php#L95)

**Pull request merged**